src/hotspot/cpu/x86/templateInterpreterGenerator_x86_64.cpp

```diff
@@ -332,6 +332,68 @@ address TemplateInterpreterGenerator::generate_CRC32C_updateBytes_entry(Abstract
   return NULL;
 }
 
+/**
+ * Method entry for static method:
+ *    java.lang.Float.float16ToFloat(short floatBinary16)
+ */
+address TemplateInterpreterGenerator::generate_Float_float16ToFloat_entry() {
+  if (!InlineIntrinsics ||
+      vmIntrinsics::is_disabled_by_flags(vmIntrinsics::_float16ToFloat) ||
+      vmIntrinsics::is_disabled_by_flags(vmIntrinsics::_floatToFloat16)) {
+    return nullptr; // Generate a vanilla entry
+  }
+  // For AVX CPUs only. f16c support is disabled if UseAVX == 0.
+  if (VM_Version::supports_f16c() || VM_Version::supports_avx512vl()) {
```

Comment:
We could check for VM_Version::supports_float16() here instead.

Reply:
Yes. And I need to remove `!InlineIntrinsics` check at line 340.